File: benchmarks/string/regex_100k.nim
import std/[times, strutils, strformat]
var lines: seq[string]
for i in 0..<100000:
  lines.add(fmt"The quick brown fox jumps over the lazy dog {i}")
let start = cpuTime()
var count = 0
for line in lines:
  for word in line.split(' '):
    if word.len >= 5:
      count += 1
echo int((cpuTime() - start) * 1_000_000)

File: benchmarks/string/regex_100k.rs
use std::time::Instant;
fn main() {
    let lines: Vec<String> = (0..100000)
        .map(|i| format!("The quick brown fox jumps over the lazy dog #{}", i))
        .collect();
    let start = Instant::now();
    let mut count = 0usize;
    for line in &lines {
        let mut in_word = false;
        let mut word_len = 0;
        for c in line.chars() {
            if c.is_alphanumeric() || c == '_' {
                word_len += 1;
                in_word = true;
            } else {
                if in_word && word_len >= 5 { count += 1; }
                word_len = 0;
                in_word = false;
            }
        }
        if in_word && word_len >= 5 { count += 1; }
    }
    println!("{}", start.elapsed().as_micros());
}
